### PR TITLE
Update changelog 0.3.1

### PR DESCRIPTION
**Pull request by sklam**

### CODE REVIEW

File: CHANGE_LOG [hunk 1]
```diff
@@ -1,7 +1,30 @@
 Version 0.4
 -----------
 
-In development
+This is a maintenace release that:
+
+* Update maximum SciPy dependency to 1.7.3.
+* Update Azure CI.
```
Comment (stuartarchibald) [inline CHANGE_LOG:7]:
```suggestion
* Updates the maximum SciPy dependency to 1.7.3.
* Updates Azure CI.
```

File: CHANGE_LOG [hunk 2]
```diff
@@ -1,7 +1,30 @@
...
+  Willard)
+* Github Actions CI
+
+Importantly, the dynamic generation of the signatures now makes numba-scipy
+compatible with multiple SciPy releases simultaneously.
+
+Contributors:
+
+* Brandon T. Willard
+* Valentin Haenel
```
Comment (stuartarchibald) [inline CHANGE_LOG:27]: Isn't this from 0.3?

Comment (sklam) [reply CHANGE_LOG]: oops, removed

File: CHANGE_LOG (hunk 1 above)
Comment (sklam) [reply CHANGE_LOG]: applied

File: CHANGE_LOG [hunk 3]
```diff
@@ -3,6 +3,21 @@ Version 0.4
 
 In development
 
+Version 0.3.1
+-------------
+
+This is a maintenace release that:
```
Comment (stuartarchibald) [inline CHANGE_LOG:9]:
```suggestion
This is a maintenance release that:
```

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch.

**Comment by sklam** [comment]
GH is incorrectly reporting about a cancelled build. Merging

**Pull request merged**